src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -630,14 +630,16 @@ assert hasLeadOnSegment(segment.getId())
             null != segment.getStartTime() ? segment.getStartTime().toDate() : null,
             segment.getFailCount()));
 
-    if (null != segment.getEndTime()) {
-      assert RepairSegment.State.DONE == segment.getState();
+    if (null != segment.getEndTime() || State.NOT_STARTED == segment.getState()) {
+      assert RepairSegment.State.RUNNING != segment.getState();
+      assert RepairSegment.State.NOT_STARTED != segment.getState() || null == segment.getEndTime();
+      assert RepairSegment.State.DONE != segment.getState() || null != segment.getEndTime();
```

Comment:
I find these tests hard to read because of the negations and the `||` operator. It's not obvious what is tested unless you give it a good thought.

Could we trade these against individual methods in the RepairSegment object ?
Something like : 
```
public boolean isNotStarted() {
  return RepairSegment.State.NOT_STARTED == getState();
}

public boolean isRunning() {
  return RepairSegment.State.RUNNING == getState();
}

public boolean isDone() {
  return RepairSegment.State.RUNNING != getState();
}


public boolean hasEndTime() {
  return null != getEndTime();
}
...
```
Then we would use those methods combined together in the assertions.
It wouldn't change much but the code intent would be easier to understand.

Reply:
yeah fair point. i'll add those methods (but just as private for now).
i'll also add assert messages, as this helps readability too…

Reply:
done. and changed the `assert` statements to `Preconditions.checkArgument(..)` as there's no performance overhead on these checks.